openbb_platform/providers/oecd/openbb_oecd/utils/helpers.py

```diff
@@ -46,3 +54,163 @@ def fetch_data(url: str, csv_kwargs: Optional[Dict] = None, **kwargs: Any) -> Da
+        for value in series.findall(".//generic:Value", namespaces=namespaces):
+            series_data[value.get("id")] = value.get("value")
+        # Extract observation values
+        for obs in series.findall("./generic:Obs", namespaces=namespaces):
+            obs_data = series_data.copy()
+            obs_data["TIME_PERIOD"] = obs.find(
+                "./generic:ObsDimension", namespaces=namespaces
+            ).get("value")
+            obs_data["VALUE"] = obs.find(
+                "./generic:ObsValue", namespaces=namespaces
+            ).get("value")
+            data.append(obs_data)
+
+    # Create a DataFrame
+    return DataFrame(data)
+
+
+def parse_url(url: str) -> DataFrame:
+    """Helper function to parse the SDMX url and return a dataframe.
+
+    Parameters
+    ----------
+    url:str
+        URL to parse
+
+    Returns
+    -------
+    DataFrame
+        Pandas dataframe containing URL data
+    """
+    response = helpers.make_request(url)
+    response.raise_for_status()
+    return oecd_xml_to_df(response.text)
+
+
+def check_cache_exists_and_valid(function: str, cache_method: str = "parquet") -> bool:
+    """Check if the cache exists and is valid.
+
+    Parameters
+    ----------
+    function : str
+        The name of the function for which the cache is being checked.
+    cache_method : str, optional
+        The method used for caching (default is 'parquet').
+
+    Returns
+    -------
+    bool
+        True if the cache exists and is valid for the current day, False otherwise.
+    """
+    # TODO: add setting to disable cache for tests
+
+    if cache_method != "parquet":
+        raise NotImplementedError("Currently only working with parquet")
+    # First check that the cache exists.  This will be a parquet and a timestamp
+    cache_path = f"{cache}/{function}.{cache_method}"
+    time_cache_path = f"{cache}/{function}.timestamp"
+    if Path(cache_path).exists() and Path(time_cache_path).exists():
+        # Now check that the cache is valid.  I am going to check that we write to a file the date the cache was made
+        # Read the timestamp
+        with open(time_cache_path) as f:
+            cached_date = f.read().strip()
+        # TODO:  More robust caching logic
+        if cached_date == str(date.today()):
+            return True
+        return False
+    return False
+
+
+def write_to_cache(function: str, data: DataFrame, cache_method: str) -> None:
+    """Write data to the cache.
+
+    Parameters
+    ----------
+    function : str
+        The name of the function for which data is being cached.
+    data : DataFrame
+        The DataFrame to be cached.
+    cache_method : str
+        The method used for caching the data.
+
+    Raises
+    ------
+    NotImplementedError
+        If the cache_method is not 'parquet'.
+    """
+    if cache_method == "parquet":
+        cache_path = f"{cache}/{function}.parquet"
+        data.to_parquet(cache_path, engine="pyarrow")
```

Comment:
"pyarrow" requires Visual C++ Redistributable - X-Code and GCC for Mac - is there an engine that can avoid this process?

Reply:
I think that's actually a parquet requirement even more than pyarrow. So if we wanted to avoid that I would go either with a different file or some error handling around it. And also the file handling should be moved into core helpers, I just did it here as a quick win.

Ultimately I'll let @IgorWounds weigh in

Reply:
I made csv the default + removed pyarrow from deps.  Idea would be that a user setting could select the file to use.